### ISSUE TITLE
gradle countExternalDebugDexMethods fails

### ISSUE THREAD

**Issue by vanniktech** [KeepSafe/dexcount-gradle-plugin#35]
> Parallel execution with configuration on demand is an incubating feature.
:app:countExternalDebugDexMethods
Total methods in app-external-debug-1.0.6.apk: 25001
Total fields in app-external-debug-1.0.6.apk:  16632
FAILURE: Build failed with an exception.
* What went wrong:
Could not add entry ':app:countExternalDebugDexMethods' to cache taskArtifacts.bin (/Users/user/tmp/project/.gradle/2.4/taskArtifacts/taskArtifacts.bin).
> Unable to store task input properties. Property 'apkOrDex' with value 'com.android.build.gradle.internal.api.ApkVariantOutputImpl_Decorated@5717906e' cannot be serialized.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
BUILD FAILED

Full stacktrace can be found here: http://pastebin.com/k2NXUjBJ

**Comment by benjamin-bader** [comment]
Interesting!  This should be fixed in the latest snapshot build (in pull request #32).  How did you reproduce this, though?  What triggers this error?

**Comment by vanniktech** [comment]
Well I do append the version number to the apk as you can see `app-external-debug-1.0.6.apk`

For that I'm using these lines

```groovy
applicationVariants.all { variant ->
    variant.outputs.each { output ->
        output.outputFile = new File(output.outputFile.parent, output.outputFile.name.replace(".apk", "-" + defaultConfig.versionName + ".apk"))
    }
}
```

**Comment by benjamin-bader** [comment]
And when you take that out, does the plugin still crash?  I'm still fuzzy on when or why Gradle tries to serialize tasks.

**Comment by vanniktech** [comment]
Hmm okay, my mistake here. I was using `0.3.0` when I was reporting that issue. #32 does indeed fix the issue and regardless of the outputFile renaming it still works. I've checked both variants (with file renaming and without) against `0.3.1-SNAPSHOT`.

Could you release `0.3.1` so that we have the fix for that? Also would it be possible to include #34?

**Comment by benjamin-bader** [comment]
I agree, we should have a release with #32 - I'll see what I can do.  Will look at #34 as well, thanks.

**Comment by vanniktech** [comment]
I'll close this one now since it was indeed fixed by #32 and I'll wait for the next release. Thanks!
